game-core/src/main/java/games/strategy/triplea/ui/FlagDrawMode.java

```diff
@@ -12,51 +12,28 @@
-    final UnitsDrawer.UnitFlagDrawMode setting =
-        UnitsDrawer.UnitFlagDrawMode.valueOf(
-            prefs.get(
-                UnitsDrawer.PreferenceKeys.DRAW_MODE.name(),
-                UnitsDrawer.UnitFlagDrawMode.SMALL_FLAG.name()));
-    UnitsDrawer.setUnitFlagDrawMode(setting, prefs);
-    UnitsDrawer.enabledFlags =
-        prefs.getBoolean(
-            UnitsDrawer.PreferenceKeys.DRAWING_ENABLED.name(), UnitsDrawer.enabledFlags);
+  /**
+   * Updates the current flag draw mode and refreshes the map so that the current flag draw mode is
+   * rendered on map.
+   */
+  public static void toggleDrawMode(
+      final UnitsDrawer.UnitFlagDrawMode drawMode, final MapPanel mapPanel) {
+    ClientSetting.unitFlagDrawMode.setValueAndFlush(drawMode);
+    mapPanel.resetMap();
+  }
 
-    // draw mode is off, toggle small flag
-    if (!prefs.getBoolean(UnitsDrawer.PreferenceKeys.DRAWING_ENABLED.name(), false)) {
-      UnitsDrawer.enabledFlags = true;
-      prefs.putBoolean(UnitsDrawer.PreferenceKeys.DRAWING_ENABLED.name(), true);
-      prefs.put(
-          UnitsDrawer.PreferenceKeys.DRAW_MODE.name(),
-          UnitsDrawer.UnitFlagDrawMode.SMALL_FLAG.name());
-      UnitsDrawer.setUnitFlagDrawMode(UnitsDrawer.UnitFlagDrawMode.SMALL_FLAG, prefs);
-      // check if small flags, if so then draw large flags
-    } else if (UnitsDrawer.UnitFlagDrawMode.SMALL_FLAG
-        .toString()
-        .equals(
-            prefs.get(
-                UnitsDrawer.PreferenceKeys.DRAW_MODE.name(),
-                UnitsDrawer.UnitFlagDrawMode.SMALL_FLAG.name()))) {
-      UnitsDrawer.enabledFlags = true;
-      prefs.putBoolean(UnitsDrawer.PreferenceKeys.DRAWING_ENABLED.name(), true);
-      prefs.put(
-          UnitsDrawer.PreferenceKeys.DRAW_MODE.name(),
-          UnitsDrawer.UnitFlagDrawMode.LARGE_FLAG.name());
-      UnitsDrawer.setUnitFlagDrawMode(UnitsDrawer.UnitFlagDrawMode.LARGE_FLAG, prefs);
-      // otherwise we had large flags, turn drawing flags off
+  /** Toggles to the next flag draw mode. */
+  static void toggleNextDrawMode(final MapPanel mapPanel) {
+    final UnitsDrawer.UnitFlagDrawMode currentDrawMode =
+        ClientSetting.unitFlagDrawMode.getValueOrThrow();
+    if (currentDrawMode == UnitsDrawer.UnitFlagDrawMode.NONE) {
+      toggleDrawMode(UnitsDrawer.UnitFlagDrawMode.SMALL_FLAG, mapPanel);
+    } else if (currentDrawMode == UnitsDrawer.UnitFlagDrawMode.SMALL_FLAG) {
+      toggleDrawMode(UnitsDrawer.UnitFlagDrawMode.LARGE_FLAG, mapPanel);
     } else {
-      UnitsDrawer.enabledFlags = false;
-      prefs.putBoolean(UnitsDrawer.PreferenceKeys.DRAWING_ENABLED.name(), false);
-      prefs.put(
-          UnitsDrawer.PreferenceKeys.DRAW_MODE.name(),
-          UnitsDrawer.UnitFlagDrawMode.LARGE_FLAG.name());
-      UnitsDrawer.setUnitFlagDrawMode(UnitsDrawer.UnitFlagDrawMode.SMALL_FLAG, prefs);
+      toggleDrawMode(UnitsDrawer.UnitFlagDrawMode.NONE, mapPanel);
     }
-    mapPanel.resetMap();
```

Comment:
Good consideration, I was wondering about maybe doing something that would cycle through the draw modes. I'm of two minds here:
(1) when the code is external to the enum, it might be better to be explicit rather than use a more complex algorithm. One should always be careful of "clever" and consider total time to understand. Explicit and simple is worth a lot.

(2)  We do have a feature envy code smell where we're interacting with the enum in a way that could be self-encapsulated. If the logic for next draw mode were encapsulated, I don't think having such code like this in an enum is that too unexpected or difficult. It also has the nice benefit to localize any changes to the enum, so we can add/remove enum values and we don't have to update external classes that use the enum.

Update is made in: 2e5cea29a; this block now looks like this:
```
  /** Toggles to the next flag draw mode. */
  static void toggleNextDrawMode(final MapPanel mapPanel) {
    final var currentDrawMode = ClientSetting.unitFlagDrawMode.getValueOrThrow();
    toggleDrawMode(currentDrawMode.nextDrawMode(), mapPanel);
  }
```